src/Management/src/Abstractions/Diagnostics/DiagnosticsManager.cs

```diff
@@ -14,30 +14,27 @@ namespace Steeltoe.Management.Diagnostics;
 public class DiagnosticsManager : IObserver<DiagnosticListener>, IDisposable, IDiagnosticsManager
 {
     private static readonly Lazy<DiagnosticsManager> AsSingleton = new(() => new DiagnosticsManager(NullLogger<DiagnosticsManager>.Instance));
+    private readonly IList<IRuntimeDiagnosticSource> _innerSources;
+    private readonly IList<EventListener> _eventListeners;
 
     private bool _isDisposed;
-    internal IDisposable ListenersSubscription;
-    internal ILogger<DiagnosticsManager> Logger;
-    internal IList<IDiagnosticObserver> InnerObservers;
-    internal IList<IRuntimeDiagnosticSource> InnerSources;
-    internal IList<EventListener> EventListeners;
+    private IDisposable _listenersSubscription;
+    private ILogger<DiagnosticsManager> _logger;
 
-    internal bool WorkerThreadShutdown;
-    internal int Started;
+    private int _started;
 
     public static DiagnosticsManager Instance => AsSingleton.Value;
 
-    public IList<IDiagnosticObserver> Observers => InnerObservers;
-
-    public IList<IRuntimeDiagnosticSource> Sources => InnerSources;
+    public IList<IDiagnosticObserver> Observers { get; }
 
     public DiagnosticsManager(IOptionsMonitor<MetricsObserverOptions> observerOptions, IEnumerable<IRuntimeDiagnosticSource> runtimeSources,
         IEnumerable<IDiagnosticObserver> observers, IEnumerable<EventListener> eventListeners, ILogger<DiagnosticsManager> logger)
     {
         ArgumentGuard.NotNull(observers);
```

Comment:
If eventListeners and runtimeSources can be null, if no instances are registered. As a practical matter by default the extension AddMetricsObservers does add them but in the future if they were to be removed, this class will work as designed. The only thing we are depending on them in this class is to hold a reference to these instances for the application life time. The IDiagnosticObserver must not be null or the code enumerating it will throw. 

I fixed the order of the null checks.

Reply:
On lines 49 and 50 we're calling `.ToList()` on `runtimeSources` and `eventListeners`, won't that throw if they're null?

Reply:
Yes - you are correct. now fixed. thanks.